File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/DateButtonDemoXml.kt
//
// DateButtonDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class DateButtonDemoXml {

    lateinit var text: TextView
    lateinit var dateButton: com.lightningkite.butterfly.views.widget.DateButton
    lateinit var timeButton: com.lightningkite.butterfly.views.widget.TimeButton
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.date_button_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        text = view.findViewById<TextView>(R.id.text)
        dateButton = view.findViewById<com.lightningkite.butterfly.views.widget.DateButton>(R.id.dateButton)
        timeButton = view.findViewById<com.lightningkite.butterfly.views.widget.TimeButton>(R.id.timeButton)
        
        
        
        return view
    }
}

File: web/src/layout/DateButtonDemoXml.ts
//
// DateButtonDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId, startupAutoResize } from 'butterfly-web/dist/views/html'
import { customViewSetDelegate } from 'butterfly-web/dist/views/CustomView'
import htmlForDefault from './date_button_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.DateButtonDemoXml
export class DateButtonDemoXml {
    xmlRoot!: HTMLElement
    text!: HTMLElement
    dateButton!: HTMLInputElement
    timeButton!: HTMLInputElement
    loadHtmlString(): string {
        return htmlForDefault
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString())
        this.xmlRoot = view
        this.text = getViewById<HTMLElement>(view, "text")
        this.dateButton = getViewById<HTMLInputElement>(view, "dateButton")
        this.timeButton = getViewById<HTMLInputElement>(view, "timeButton")
        return view
    }
}
